### ISSUE TITLE
EDDN: Misc. /upload/ errors

### ISSUE THREAD

**Issue by Athanasius** [EDCD/EDMarketConnector#1403]
- [x] `Failed Validation "[<ValidationError: "'SystemAddress' is a required property">]" (1764, "E:D Market Connector [Windows]", "5.2.4", "https://eddn.edcd.io/schemas/journal/1", "Location")`
- [x] `Failed Validation "[<ValidationError: "{'type': ['array', 'boolean', 'integer', 'number', 'null', 'object', 'string']} is not allowed for u'\\u0421\\u0435\\u043b\\u044c\\u0441\\u043a\\u043e\\u0435 \\u0445\\u043e\\u0437\\u044f\\u0439\\u0441\\u0442\\u0432\\u043e'">]" (4188, "E:D Market Connector [Windows]", "5.2.4", "https://eddn.edcd.io/schemas/journal/1", "Location")`

- [x] `Failed Validation "[<ValidationError: "{'type': ['array', 'boolean', 'integer', 'number', 'null', 'object', 'string']} is not allowed for u'\\u0421\\u0447\\u0430\\u0441\\u0442\\u044c\\u0435'">]" (2933, "E:D Market Connector [Windows]", "5.2.4", "https://eddn.edcd.io/schemas/journal/1", "FSDJump")`
- [x] `Failed Validation "[<ValidationError: "None is not of type 'boolean'">]" (2505, "E:D Market Connector [Windows]", "5.2.4", "https://eddn.edcd.io/schemas/journal/1", "FSDJump")`

- [x] `Failed Validation "[<ValidationError: "'SystemAddress' is a required property">]" (566, "E:D Market Connector [Windows]", "5.2.4", "https://eddn.edcd.io/schemas/journal/1", "Docked")`
- [x] `Failed Validation "[<ValidationError: "{'type': ['array', 'boolean', 'integer', 'number', 'null', 'object', 'string']} is not allowed for u'\\u0412\\u044b\\u0441\\u043e\\u043a\\u0438\\u0435 \\u0442\\u0435\\u0445\\u043d\\u043e\\u043b\\u043e\\u0433\\u0438\\u0438'">]" (1124, "E:D Market Connector [Windows]", "5.2.4", "https://eddn.edcd.io/schemas/journal/1", "Docked")`
- [x] `Failed Validation "[<ValidationError: "None is not of type 'boolean'">]" (1138, "E:D Market Connector [Windows]", "5.2.4", "https://eddn.edcd.io/schemas/journal/1", "Docked")`

- [x] `Failed Validation "[<ValidationError: "None is not of type 'boolean'">]" (439, "E:D Market Connector [Windows]", "5.2.4", "https://eddn.edcd.io/schemas/fssdiscoveryscan/1", "-")`

I'd hazard a guess that the `Failed Validation "[<ValidationError: "None is not of type 'boolean'">]"` is a bad horizons or odyssey flag.

**Comment by Athanasius** [comment]
`Failed Validation "[<ValidationError: "'SystemAddress' is a required property">]" (1764, "E:D Market Connector [Windows]", "5.2.4", "https://eddn.edcd.io/schemas/journal/1", "Location")`

I just can not see how we'd ever fail to send *at least* `"SystemAddress":null` to EDDN, which would elicit `FAIL: Schema Validation: [<ValidationError: "None is not of type 'integer'">]` not `is a required property`.

I've checked throughout monitor.py and plugins/eddn.py for:

1. How 'SystemAddress' values are stored and then used.
2. Any `del ` that might be relevant.
3. Any `.pop(` that might be relevant.

and I just can't see how this would happen.

**Comment by Athanasius** [comment]
**Any and all of those EDDN validation failures could be from EDMC's EDDN replaylog (re-)attempting old messages.**

It could have been only once, e.g. when someone got back from exploration and "Delay sending until docked" was in play.

**Comment by Athanasius** [comment]
Closing in favour of #1431

**Comment by Athanasius** [comment]
Re: `Failed Validation "[<ValidationError: "None is not of type 'boolean'">]" (2505, "E:D Market Connector [Windows]", "5.2.4", "https://eddn.edcd.io/schemas/journal/1", "FSDJump")` ... I just checked:

1. The Horizons flag is now always in the `LoadGame` event.
2. The Odyssey flag is *not* present in the `LoadGame` event on Base Game or Horizons (both use the Horizons install/client).
3. However we've only ever had monitor.py tracking the Odyssey flag in a manner where it will default to `False`, never `None`.  Ref: 286060a796da3aad1718ee056b490a34c8041ae8
4. FSDJump is in the journal/1 schema, and the only fields declared to be of type boolean are the horizons and odyssey flags.

So I'm still at a loss as to how this validation error could be due to either of those flags.  Has to be an old message being replayed, right ?
